src/api/controllers/lines.js

```diff
@@ -128,17 +128,85 @@ function getStatsByEmployerId(req, res) {
             "message": response.unauthorized
         });
     } else {
-        Line.count({ employer_id: req.query.employer_id }, function (err, count) {
-            if (err) {
-                return res.send(err);
-            }
+        //use promises to have queries run async. in parallel.
+        var size = getSizeByEmployerId(req.query.employer_id);
```

Comment:
wow good organization. i didn't think of this.

Reply:
i got the idea from this site i was looking at when learning about callbacks:
http://callbackhell.com/